spec/hangman_rules_spec.rb

```diff
@@ -1,57 +1,98 @@
+    expect(guesses[:wrong_guesses]).to eq(["c"])
   end
 
-  it "does ignore letters not correctly guessed" do
-    hangman_rules = new_hangman_rules(word_generator_stub)
+  it "removes letter guessed from guessable letters" do
+    hangman_rules = new_hangman_rules("hello", new_guesses)
+    letters = guessable_letters
 
-    hangman_rules.guess("c")
+    hangman_rules.guess("a", letters)
 
-    expect(hangman_rules.right_guesses).not_to include("c")
+    expect(letters).not_to include("a")
   end
 
-  it "removes letter guessed from guessable letters" do
-    hangman_rules = new_hangman_rules(word_generator_stub)
+  it "knows when game is over and player won" do
+    guesses = new_guesses
+    hangman_rules = new_hangman_rules("hi", guesses)
+
+    hangman_rules.guess("h", guessable_letters)
+    hangman_rules.guess("i", guessable_letters)
+
+    expect(hangman_rules.game_state).to eq(:won)
+  end
+
+  it "does not consider doubles in secret word when checking if won" do
+    guesses = new_guesses
+    hangman_rules = new_hangman_rules("hh", guesses)
+
+    hangman_rules.guess("h", guessable_letters)
 
-    hangman_rules.guess("a")
+    expect(hangman_rules.game_state).to eq(:won)
+  end
+
+  it "knows player lost if they guess 11 times wrongly" do
+    guesses = new_guesses
+    hangman_rules = new_hangman_rules("hi", guesses)
+
+    guess_wrong_eleven_times(hangman_rules)
 
-    expect(hangman_rules.available_letters).not_to include("a")
+    expect(hangman_rules.game_state).to eq(:lost)
   end
 
-   it "returns all letters of alphabet" do
-    hangman_rules = new_hangman_rules(word_generator)
+  it "knows if game is not over" do
+    guesses = new_guesses
+    hangman_rules = new_hangman_rules("hi", guesses)
 
-    alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
+    hangman_rules.guess("h", guessable_letters)
 
-    expect(hangman_rules.available_letters).to eq(alphabet)
+    expect(hangman_rules.game_state).to eq(:ongoing)
   end
 
   private
 
-  def new_hangman_rules(word_generator)
-    HangmanRules.new(word_generator)
+  def new_hangman_rules(word, right_guesses)
+    HangmanRules.new(word, right_guesses)
+  end
+
+  def guessable_letters
+    ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
+  end
+
+  def new_guesses
+    {:right_guesses => [], :wrong_guesses => []}
+  end
+
+  def guess_wrong_eleven_times(hangman_rules)
+    hangman_rules.guess("e", guessable_letters)
+    hangman_rules.guess("f", guessable_letters)
+    hangman_rules.guess("a", guessable_letters)
+    hangman_rules.guess("b", guessable_letters)
+    hangman_rules.guess("c", guessable_letters)
+    hangman_rules.guess("d", guessable_letters)
+    hangman_rules.guess("g", guessable_letters)
+    hangman_rules.guess("z", guessable_letters)
```

Comment:
Instead of repeating this line 11 times, could you iterate over an array of the letters and for each call the method?